crates/sui-indexer-alt-graphql/src/api/mutation.rs

```diff
@@ -1,9 +1,28 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use anyhow::anyhow;
 use async_graphql::{Context, Object, Result};
+use sui_indexer_alt_reader::fullnode_client::Error::GrpcExecutionError;
+use sui_indexer_alt_reader::fullnode_client::FullnodeClient;
 
-use crate::{api::types::transaction_execution_input::TransactionExecutionInput, error::RpcError};
+use sui_types::signature::GenericSignature;
+use sui_types::transaction::TransactionData;
+
+use crate::{
+    api::{scalars::base64::Base64, types::execution_result::ExecutionResult},
+    error::{bad_user_input, RpcError},
+};
+
+/// Error type for user input validation in executeTransaction
+#[derive(thiserror::Error, Debug)]
+enum TransactionInputError {
+    #[error("Invalid BCS encoding in transaction data: {0}")]
+    InvalidTransactionBcs(String),
+
+    #[error("Invalid BCS encoding in signature {index}: {message}")]
+    InvalidSignatureBcs { index: usize, message: String },
```

Comment:
nit (optional): You might consider keeping the error written in terms of error types instead of forcing the conversion into strings. In this case, it's not a big deal either way, but the pattern is a bit of a codesmell (an error accepting another error as a string).

```suggestion
    #[error("Invalid BCS encoding in transaction data: {0}")]
    InvalidTransactionBcs(bcs::Error),

    #[error("Invalid BCS encoding in signature {index}: {err}")]
    InvalidSignatureBcs { index: usize, err: bcs::Error },
```

Reply:
SG. Done